apps/music/test/marionette/playlist_test.js

```diff
@@ -30,6 +30,30 @@ marionette('Music player playlist', function() {
 
   });
 
+
+  suite('Special names', function() {
+    setup(function() {
+      client.fileManager.removeAllFiles();
+      client.fileManager.add([
+        // Title: "dump 2>&1 < ~/®"
+        {
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/d.ogg'
+        }
+      ]);
+    });
+
+    test('Check name with >, <, ~, & and some Unicode', function() {
+      music.launch();
+      music.waitForFirstTile();
+      music.switchToSongsView();
+
+      // this will wait on the first song as well.
+      var song = music.firstSong;
+      assert.equal(PlaylistHelper.mainTitle(song), 'dump 2>&1 < ~/®');
```

Comment:
This song title goes well with the other ones on the fake album. 
But it isn't sufficient for this test because the angle brackets and the & are not close enough to real HTML.
If I understand it correctly, the point of this test is to make sure that we are not setting song titles on innerHTML (or are properly escaping the HTML if we do).  But the HTML parser is smart enough to figure out that this song title is not HTML. So, consider this code:

```
document.body.innerHTML = 'dump 2>&1 < ~/®';
assert.equal(document.body.textContent, 'dump 2>&1 < ~/®')
```

That assertion passes even though the innerHTML contained unescaped special characters.

For a valid test, I think you need a song title like 

```
<b.&amp; Injection Vulnerabliity</b>
```